### DISCUSSION TITLE
Bionic Reading Mode

### DISCUSSION THREAD

**Discussion by Ioulan** [johnfactotum/foliate#1223] [Ideas]
> `Bionic Reading` is a font tool that guides the eyes through artificial fixation points to make reading easier and faster
 <sub>src: [howtogeek](https://www.howtogeek.com/807981/what-is-bionic-reading-and-how-do-you-use-it/)</sub>

What do you guys think?
Foliate having this kind of feature?
Will this be beneficial for the user? or will this be pain to develop for the maintainer?

**D**o **yo**u **fee**l like **yo**u can **rea**d **thi**s **senten**ce **mor**e **eas**ily than **th**e **othe**rs?
https://bionic-reading.com/

**Comment by johnfactotum** [comment]
In general, this kind of feature probably makes more sense as a plugin/extension/userscript.

Specifically, Bionic Reading seem to rely on a proprietary algorithm which I'm not sure how it's suppose to work.

Alternatively, one could use a tool to preprocess the book to enable Bionic Reading in any reading system, though this has the slight disadvantage that it would break CFIs (unlike implementations within the reading system where it could ignore the extra nodes).

**Comment by matkoniecz** [comment]
> Do you feel like you can read this sentence more easily than the others?

no

**Reply by khlsvr** [reply]
Same. At first, reading that line here seemed harder or more irritating to read than regular text, but I checked the howtogeek link and read some more lines there and it already felt equal first and then better for eyes. Maybe the eyes don't need to try so much once they adapt to it or that's my early impression now.

**Comment by slumbergod** [comment]
There are several web extensions for browsers now so I guess it's not quite as proprietary as it was when it first hit the web. 

I actually came here specifically to see if there were plans to add an extension for it. I find a wall of text very tiring and as weird as it looks, "bionic" formatting really speeds it up for me. I hope it is possible in the future even it it is just importing a chrome or firefox extension. For now I will see if I can find an ebook extension for the browser than supports it.

**Reply by khlsvr** [reply]
With Fast Font https://github.com/Born2Root/Fast-Font I can get it working on Foliate. You could add bionic reading to any font using the technique they use on that github page, but it would involve creating a new bionic version of the font with help of a script or font programs.

The font itself transforms first chars to bold ones using contextual alternates
